### PR TITLE
testcase/filesystem: Fix a problem with procfs in the file system testcase

### PR DESCRIPTION
**Pull request by KimDongEon**
1. Failing to test procfs on file system tc without mounting procfs fixes this problem.

2. To test the procfs mount, unmount the existing automount

### CODE REVIEW

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 1]
```diff
@@ -267,6 +267,9 @@ void tc_fs_procfs_main(void)
 	int ret;
 	struct stat st;
 
+	ret = umount("/proc");
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: please change it to PROC_MOUNTPOINT

Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]:
```
#ifdef CONFIG_FS_AUTOMOUNT_PROCFS
...
#endif
```

File: build/configs/cy4390x/tc/defconfig [hunk 2]
```diff
@@ -1076,6 +1075,8 @@ CONFIG_KERNEL_CMDS=y
 CONFIG_FS_CMDS=y
 CONFIG_FSCMD_BUFFER_LEN=64
 CONFIG_NET_CMDS=y
+CONFIG_NET_PING_CMD=y
```
Comment (jeongarmy) [inline build/configs/cy4390x/tc/defconfig]: Please exclude something which is not related to procfs from this commit.

**Comment by seinfra** [comment]
Target : [f00ffe622177507677f7c796de3df82e1bf1f8f8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f00ffe622177507677f7c796de3df82e1bf1f8f8] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 3]
```diff
@@ -267,6 +267,9 @@ void tc_fs_procfs_main(void)
 	int ret;
 	struct stat st;
 
+	/* This is a call to ensure unmount status, not test purpose. */
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: please add config regarding auto mount & TC_ASSERT to check ret.

File: build/configs/cy4390x/tc/defconfig (hunk 2 above)
Comment (KimDongEon) [reply build/configs/cy4390x/tc/defconfig]: I modified it through Kconfig without modifying defconfig directly.
File Systems -> Auto mount The PROCFS File System was turned on,
The result is another change.
Is it correct to edit only the auto mount by hand?

**Comment by Taejun-Kwon** [comment]
Please modify fully what I mentioned.
1. Match proc dev name fs_initialize.c & proc test file
2. Move Procfs features from fs_main.c to tc_fs_procfs.c
3. Please add 'select' or 'depend on' regarding CONFIG_FS_PROCFS in CONFIG_FS_AUTOMOUNT_PROCFS's Kconfig

**Comment by seinfra** [comment]
Target : [226de65fb44248e8faf728fabe81bd4846501aba] - Code Rule Check (C++) OK.

File: build/configs/cy4390x/tc/defconfig (hunk 2 above)
Comment (KimDongEon) [reply build/configs/cy4390x/tc/defconfig]: modify!

**Comment by seinfra** [comment]
Target : [419e91adc22cf0d6575489bd5e3aa766a74fd200] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [419e91adc22cf0d6575489bd5e3aa766a74fd200] - Code Rule Check OK.

File: build/configs/artik053/tc/defconfig [hunk 4]
```diff
@@ -893,6 +893,7 @@ CONFIG_SMARTFS_ALIGNED_ACCESS=y
 # CONFIG_SMARTFS_JOURNALING is not set
 # CONFIG_SMARTFS_SECTOR_RECOVERY is not set
 CONFIG_FS_PROCFS=y
+CONFIG_FS_AUTOMOUNT_PROCFS=y
```
Comment (jeongarmy) [inline build/configs/artik053/tc/defconfig]: Please copy and paste configs in "SMARTFS options" only made by menuconfig automatically.

**Comment by seinfra** [comment]
Target : [3b1b49a0dfdbcdf71b031b11ce3b714868382c78] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3b1b49a0dfdbcdf71b031b11ce3b714868382c78] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6b4e4e99b089bcf87cebefdb9d46e2c1fea297cc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6b4e4e99b089bcf87cebefdb9d46e2c1fea297cc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [afe9a1ff6e412c93e76ad7f2df8d9d24879c8e87] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [afe9a1ff6e412c93e76ad7f2df8d9d24879c8e87] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c [hunk 5]
```diff
@@ -266,8 +266,15 @@ void tc_fs_procfs_main(void)
...
 	struct stat st;
-
+	
 	ret = mount(NULL, PROC_MOUNTPOINT, "procfs", 0, NULL);
+	if (ret == EEXIST) {
+		ret = umount(PROC_MOUNTPOINT);
+		TC_ASSERT_EQ("umount", ret, OK);
+
+		ret = mount(NULL, PROC_MOUNTPOINT, "procfs", 0, NULL);
+	}
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: I think this is not right with respect to procfs automount. you should mount procfs only when `CONFIG_FS_AUTOMOUNT_PROCFS` is disabled. 
i.e.
```
#ifndef CONFIG_FS_AUTOMOUNT_PROCFS
    ret = mount(NULL, PROC_MOUNTPOINT, "procfs", 0, NULL);
    .
    .
#endif
```

Comment (lokeshbv) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c]: When procfs automount is enabled, User assume that there will not be "/proc" mount point. But when `CONFIG_FS_AUTOMOUNT_PROCFS`  is disabled and fs tc is executed, "/proc" remains mounted. Is it right?

**Pull request merged**